### PR TITLE
DXI- 29263

### PR DESCRIPTION
**Pull request by staware30**
JIRA: https://jira.corp.adobe.com/browse/DXI-29263

- Fetching asset count based on Deliverable Type and Platform
- Both metadata types are now interactive buttons
- Displaying the total asset count upon click

Design Doc: 
- https://xd.adobe.com/view/4f7f6edd-3037-4902-8d37-b51e7a79b543-1a2c/screen/d33e1ab8-2c8f-4d7f-91a1-ccb13839a2b5?fullscreen

Test URLs:
- Before: https://main--assets-distribution-portal--adobe.hlx.page/sample-public-site
- After: https://dxi-29263--adobe-gmo--hlxsites.hlx.page/qa/program-details?programName=ENV%20Update%20Test%20Campaign&programID=671a910f0005d2f623434630bf23c09f&path=/content/dam/gmo-cf/en/digital-media-dme/programs/env-update-test-campaign-671a910f0005d2f623434630bf23c09f

### CODE REVIEW

File: blocks/gmo-program-details/gmo-program-details.css [hunk 1]
```diff
@@ -45,6 +45,13 @@ body {
     }
     & .h3 {
         font: normal normal bold 14px/21px Adobe Clean;
+        /* &.clear-selection{
+            font-weight: normal;
+            color: #0096FF;
+            cursor: pointer;
+            
+            margin-left: 5px;
+        } */
```
Comment (mdickson-adbe) [inline blocks/gmo-program-details/gmo-program-details.css:54]: @staware30 Can we remove the commented CSS?

File: blocks/gmo-program-details/gmo-program-details.js [hunk 2]
```diff
@@ -309,14 +315,80 @@ async function addProgramStats(block) {
...
+    const deliverableTypeToPlatformsMap = new Map();
+    // Create a map to store the associations
+    const platformToDeliverableTypesMap = new Map();
+
+    // Populate the map
+    programData.data.deliverableList.items.forEach(item => {
+        const deliverableType = item.deliverableType; //deliverableType
+        const platforms = item.platforms; // array of platform strings
+
+        // if (!deliverableTypeToPlatformsMap.has(deliverableType)) {
```
Comment (mdickson-adbe) [inline blocks/gmo-program-details/gmo-program-details.js:336]: Let's remove the commented code

File: blocks/gmo-program-details/gmo-program-details.js [hunk 3]
```diff
@@ -309,14 +315,80 @@ async function addProgramStats(block) {
...
+            //     deliverableTypesPlatformAssetCountMap.set(platform, assetResult.assetCount);
+            //     // Create a new set for each platform with the deliverable type added to the new array.
+            //     platformToDeliverableTypesMap.set(platform, new Array());
+            // }
+            // platformToDeliverableTypesMap.get(platform).push(deliverableType);
+        });
+
+    });
+
+    console.log('Deliverable Type to Platforms Map:', deliverableTypeToPlatformsMap);
```
Comment (mdickson-adbe) [inline blocks/gmo-program-details/gmo-program-details.js:385]: Would suggest pulling the console statements out also unless absolutely necessary

**Review by mdickson-adbe — COMMENTED** [review]
Looks good functionally but we should probably clean up the commented code and log statements.

**Review by mdickson-adbe — APPROVED** [review]

File: blocks/gmo-program-details/gmo-program-details.js [hunk 4]
```diff
@@ -452,20 +504,114 @@ function switchTab(tab) {
...
+                    allTags.forEach(t => {
+                    t.style.display = 'inline-block';
+                     t.classList.remove('selected'); // Remove the selected class
+                    });
+                    const associatedItems = associationMap.get(scope);
+
+                    // Reset the associated buttons
+                    associatedItems.forEach(async (_count, key) => {
+                        const associatedTag = Array.from(allTags).find(t => t.id.includes(key));
+                        // console.log('Associated Tag:', associatedTag);
```
Comment (shiv-gup) [inline blocks/gmo-program-details/gmo-program-details.js:536]: @staware30 remove console.log statment

**Review by shiv-gup — APPROVED** [review]

**Pull request merged**